### PR TITLE
refactor(compare_map_segmentation): rework parameters

### PR DESCRIPTION
**Pull request by yuntianyi-chen**
## Description

Refactor the ROS Node configuration layout for the compare_map_segmentation package according to the [guideline](https://github.com/orgs/autowarefoundation/discussions/3433).
- Create the schema and parameter files.
- Remove the default values from the source code to avoid duplication.
- Update the launch files.

## Related links

None

## Tests performed

The refactored node was successfully built and tested by launching this node individually. 

## Notes for reviewers

`voxel_grid_map_loader.cpp` contains declare_parameter functions but is not registered in `CmakeLists.txt`. I'm not sure if I should create param and schema files for it.

## Interface changes

- Create 5 param files and 5 schema files.
- Parameter values are now required to be passed to the node when launched.

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
LGTM
@miursh @badai-nguyen could you review this PR?

**Comment by yuntianyi-chen** [comment]
There is one thing worth noticing. `voxel_grid_map_loader.cpp` contains declare_parameter functions but is not registered in `CmakeLists.txt`. I'm not sure if I should create param and schema files for it.

**Comment by badai-nguyen** [comment]
> There is one thing worth noticing. `voxel_grid_map_loader.cpp` contains declare_parameter functions but is not registered in `CmakeLists.txt`. I'm not sure if I should create param and schema files for it.

@yuntianyi-chen The paramters declared in `voxel_grid_map_loader.cpp` are common for all voxel_based filter, and I think it is better to define these parameter in each param.yaml for each voxel_based filter.

File: perception/compare_map_segmentation/launch/voxel_based_compare_map_filter.launch.xml [hunk 1]
```diff
@@ -10,16 +12,13 @@
...
   <node pkg="compare_map_segmentation" exec="voxel_based_compare_map_filter_node" name="voxel_based_compare_map_filter_node" output="screen">
     <remap from="input" to="$(var input)"/>
     <remap from="map" to="$(var input_map)"/>
     <remap from="output" to="$(var output)"/>
-    <param name="distance_threshold" value="$(var distance_threshold)"/>
-    <param name="downsize_ratio_z_axis" value="$(var downsize_ratio_z_axis)"/>
-    <param name="use_dynamic_map_loading" value="$(var use_dynamic_map_loading)"/>
     <param name="timer_interval_ms" value="$(var timer_interval_ms)"/>
     <param name="map_update_distance_threshold" value="$(var map_update_distance_threshold)"/>
     <param name="map_loader_radius" value="$(var map_loader_radius)"/>
```
[the remark this answers is not in the document]
Comment (yuntianyi-chen) [reply perception/compare_map_segmentation/launch/voxel_based_compare_map_filter.launch.xml]: Moved.

**Comment by yuntianyi-chen** [comment]
> > There is one thing worth noticing. `voxel_grid_map_loader.cpp` contains declare_parameter functions but is not registered in `CmakeLists.txt`. I'm not sure if I should create param and schema files for it.
> 
> @yuntianyi-chen The paramters declared in `voxel_grid_map_loader.cpp` are common for all voxel_based filter, and I think it is better to define these parameter in each param.yaml for each voxel_based filter.

It seems like the parameters only occur in `voxel_based_compare_map_filter`. I didn't find them in other `voxel_based` filters. So I just added these parameters in the param.yaml and schema.json files for this filter.

**Review by badai-nguyen — APPROVED** [review]
LGTM

**Comment by kminoda** [comment]
@badai-nguyen @yuntianyi-chen Could you merge this PR if it's ready?

**Review by badai-nguyen — COMMENTED** [review]
> > @yuntianyi-chen The paramters declared in voxel_grid_map_loader.cpp are common for all voxel_based filter, and I think it is better to define these parameter in each param.yaml for each voxel_based filter.
> 
> It seems like the parameters only occur in voxel_based_compare_map_filter. I didn't find them in other voxel_based filters. So I just added these parameters in the param.yaml and schema.json files for this filter.

@yuntianyi-chen 
I forgot one point, the config/xxx.param.yaml you added is mainly for standalone execution of the node but currently it thrown error because of missing params. 
For example
```
ros2 launch compare_map_segmentation voxel_distance_based_compare_map_filter.launch.xml 
[INFO] [launch]: All log files can be found below /home/autoware/.ros/log/2023-11-30-09-38-04-330020-npc2201009-ubuntu22-35681
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [voxel_distance_based_compare_map_filter_node-1]: process started with pid [35701]
[voxel_distance_based_compare_map_filter_node-1] [INFO] [1701304684.625355877] [voxel_distance_based_compare_map_filter_node]: Filter (as Component) successfully created with the following parameters:
[voxel_distance_based_compare_map_filter_node-1]  - approximate_sync : false
[voxel_distance_based_compare_map_filter_node-1]  - use_indices      : false
[voxel_distance_based_compare_map_filter_node-1]  - latched_indices  : false
[voxel_distance_based_compare_map_filter_node-1]  - max_queue_size   : 5
[voxel_distance_based_compare_map_filter_node-1] terminate called after throwing an instance of 'rclcpp::exceptions::UninitializedStaticallyTypedParameterException'
[voxel_distance_based_compare_map_filter_node-1]   what():  Statically typed parameter 'map_loader_radius' must be initialized.
[ERROR] [voxel_distance_based_compare_map_filter_node-1]: process has died [pid 35701, exit code -6, cmd '/home/autoware/autoware/install/compare_map_segmentation/lib/compare_map_segmentation/voxel_distance_based_compare_map_filter_node --ros-args -r __node:=voxel_distance_based_compare_map_filter_node --params-file /home/autoware/autoware/install/compare_map_segmentation/share/compare_map_segmentation/config/voxel_distance_based_compare_map_filter.param.yaml -r input:=/input -r map:=/map -r output:=/output'].

```
It would be very appreciated if you could add some missed parameters as suggestion.

**Pull request merged**